syncano/connection.py

```diff
@@ -234,7 +234,12 @@ def make_request(self, method_name, path, **kwargs):
         :raises SyncanoValueError: if invalid request method was chosen
         :raises SyncanoRequestError: if something went wrong during the request
         """
-        files = kwargs.get('data', {}).pop('files', None)
+        data = kwargs.get('data', {})
+
+        files = {k: v for k, v in data.iteritems()
+                 if hasattr(v, 'read')}
```

Comment:
Well. In such case we can have in files some object which is not a file at all.

It rather be rare situation - but can happen.